app/scripts/controllers/sign.ts

```diff
@@ -60,11 +62,15 @@ const getDefaultState = () => ({
 
 export type CoreMessage = AbstractMessage & {
   messageParams: AbstractMessageParams;
+  securityProviderResponse?: Record<string, Json> | undefined;
 };
 
-export type StateMessage = Required<AbstractMessage> & {
+export type StateMessage = Omit<
+  Required<AbstractMessage>,
+  'securityProviderResponse'
```

Comment:
I'm confused why we are removing the `securityProviderResponse` property then redefining it. If it's part of `AbstractMessage`, we shouldn't have to redefine here right?